### ISSUE TITLE
Layout/RedundantLineBreak misses case inside block

### ISSUE THREAD

**Issue by mockdeep** [rubocop/rubocop#10311]
## Expected behavior

Given the below code, I would expect the `Layout/RedundantLineBreak` rule to report that the line inside the block should not be wrapped. **Note:** our line length is set to 84 and `InspectBlocks` is set to `false`.

```rb
some_array.map do |something|
  my_slightly_long_but_not_that_long_method(
    something,
  )
end
```

The code should be enforced to not wrap:
```rb
some_array.map do |something|
  my_slightly_long_but_not_that_long_method(something)
end
```

## Actual behavior

Rubocop does not report an error. If I add *another* line to the block it does report an issue:

```rb
some_array.map do |something|
  foo
  my_slightly_long_but_not_that_long_method(
    something,
  )
end

# C: [Correctable] Layout/RedundantLineBreak: Redundant line break detected.                                                                                                
#  my_slightly_long_but_not_that_long_method( ...
#  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## RuboCop version

```
$ rubocop -V
1.23.0 (using Parser 3.0.3.2, rubocop-ast 1.15.0, running on ruby 2.7.5 x86_64-linux)
  - rubocop-performance 1.12.0
  - rubocop-rails 2.12.4
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.6.0
```
